marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/reader/ReadDataWithCustomEndpointsTest.java

```diff
@@ -0,0 +1,46 @@
+package com.marklogic.hub.spark.sql.sources.v2.reader;
+
+import com.marklogic.client.document.GenericDocumentManager;
+import com.marklogic.client.io.DocumentMetadataHandle;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.InputStreamHandle;
+import com.marklogic.hub.spark.sql.sources.v2.Options;
+import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+public class ReadDataWithCustomEndpointsTest extends AbstractSparkReadTest {
+
+    @BeforeEach
+    void installCustomJobEndpoints() {
+        GenericDocumentManager mgr = getHubClient().getModulesClient().newDocumentManager();
+        DocumentMetadataHandle metadata = new DocumentMetadataHandle()
+            .withPermission("data-hub-operator", DocumentMetadataHandle.Capability.READ, DocumentMetadataHandle.Capability.UPDATE, DocumentMetadataHandle.Capability.EXECUTE);
+
+        String apiPath = "/custom-job-endpoints/initializeRead.api";
```

Comment:
Let's put these in a folder called "/custom-read-endpoints" so it's clear that they're for testing the read endpoint.